### PR TITLE
[SOL-371] Stepper API integration

### PR DESCRIPTION
**Pull request by ilie-lupu-ts**
## Added
- Log the name of redux action
- Added AuthType.onboarding and removed the AuthErrorType.incompleteOnboarding (couldn't be used because auth error doesn't have a cognito user, which we need to make the request to `/signup/progress`)
- When a user goes back from the Stepper Screen after login, he will be redirected to the WelcomeScreen

## Screenshots

https://github.com/ivoryio/Ivory/assets/127083262/2d0ebf07-d697-4dbf-8e14-24f3ac437d08

### CODE REVIEW

File: lib/infrastructure/onboarding/onboarding_service.dart [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+import 'package:solarisdemo/models/onboarding/onboarding_progress.dart';
+import 'package:solarisdemo/models/onboarding/onboarding_service_error_type.dart';
+import 'package:solarisdemo/models/user.dart';
+import 'package:solarisdemo/services/api_service.dart';
+
+class OnboardingService extends ApiService {
+  OnboardingService({super.user});
+
+  Future<OnboardingServiceResponse> getOnboardingProgress({required User user}) async {
+    if (this.user == null) {
```
Comment (Tudor90) [inline lib/infrastructure/onboarding/onboarding_service.dart]: Daca userul nu e optional si e required nu cred ca mai ai nevoie de if-ul asta

File: lib/redux/onboarding/onboarding_progress_reducer.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import 'package:solarisdemo/redux/onboarding/onboarding_progress_action.dart';
+
+import 'onboarding_progress_state.dart';
+
+OnboardingProgressState onboardingProgressReducer(OnboardingProgressState currentState, dynamic action) {
+  if (action is GetOnboardingProgressCommandAction) {
```
Comment (Tudor90) [inline lib/redux/onboarding/onboarding_progress_reducer.dart]: nu cred ca inteleg de ce tre' sa schimbi state-ul pt actionul asta

Comment (ilie-lupu-ts) [reply lib/redux/onboarding/onboarding_progress_reducer.dart]: Am pus ca sa pot afisa un loading pe stepper screen. 
Daca nu schimb state-ul, cand userul ajunge **din nou** pe stepper screen o sa vada informatiile care sunt deja in state (in timpul asta facandu-se un request in spate care o sa schimbe state-ul)

File: lib/infrastructure/onboarding/onboarding_service.dart (hunk 1 above)
Comment (ilie-lupu-ts) [reply lib/infrastructure/onboarding/onboarding_service.dart]: Am modificat

**Review by Tudor90 — APPROVED** [review]

**Pull request merged**